adapters/beachfront/beachfront.go

```diff
@@ -197,289 +184,327 @@ func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
 	return reqs, errs
 }
 
-func preprocess(request *openrtb.BidRequest) (beachfrontReqs beachfrontRequests, errs []error) {
-	var videoImps = make([]openrtb.Imp, 0)
-	var bannerImps = make([]openrtb.Imp, 0)
+func testForValidBanner(request *openrtb.BidRequest, index int) bool {
+	if request.Imp[index].Banner != nil && ((request.Imp[index].Banner.Format != nil &&
+
+		len(request.Imp[index].Banner.Format) != 0 &&
+		request.Imp[index].Banner.Format[0].H != 0 &&
+		request.Imp[index].Banner.Format[0].W != 0) ||
+
+		(request.Imp[index].Banner.H != nil &&
+			request.Imp[index].Banner.W != nil)) {
+
+		return true
+	}
+
+	return false
+}
+
+func preprocess(request *openrtb.BidRequest, errs *[]error) (beachfrontReqs requests) {
+	var videoImps = make([]openrtb.Imp, 0, len(request.Imp))
+	var gotBanner bool
 
 	for i := 0; i < len(request.Imp); i++ {
-		if request.Imp[i].Banner != nil && ((request.Imp[i].Banner.Format[0].H != 0 && request.Imp[i].Banner.Format[0].W != 0) ||
-			(request.Imp[i].Banner.H != nil && request.Imp[i].Banner.W != nil)) {
-			bannerImps = append(bannerImps, request.Imp[i])
+		if !gotBanner {
+			gotBanner = testForValidBanner(request, i)
 		}
 
 		if request.Imp[i].Video != nil {
-			videoImps = append(videoImps, request.Imp[i])
+			imp := request.Imp[i]
+			var secure int8 = 0
+
+			if request.Site != nil && request.Site.Page != "" {
+				secure = isPageSecure(request.Site.Page)
+				imp.Secure = &secure
+			} else {
+				// &secure = 0 at this point
+				if imp.Secure != nil && imp.Secure != &secure {
+					imp.Secure = request.Imp[i].Secure
```

Comment:
How are `imp.Secure` and `request.Imp[i].Secure` going to be different here? L214 assigns `imp` to be `request.Imp[i]` so they should be the same, no?

Reply:
Right. But also, if there is no value for page in the beachfront backend request, it will return null (204). RTB will allow request.Site.Domain or request.Site.Id, but beachfront won't. So now (still locally), if request.Site.Page is not set, I'm just dropping the imp.

Reply:
Hum. But I gotta check for App too.